osxkeychain.go

```diff
@@ -177,11 +199,28 @@ func FindGenericPassword(attributes *GenericPasswordAttributes) ([]byte, error)
 	defer C.free(unsafe.Pointer(accountName))
 
 	var passwordLength C.UInt32
-
 	var password unsafe.Pointer
+	var keychainRef C.CFTypeRef
+
+	// search specific keychains, otherwise the defaults are used
+	if len(attributes.Keychain) > 0 {
```

Comment:
ah, is this the code duplication you were talking about?

I'd probably do something like write a function that takes an array of strings and returns a CFTypeRef (or better yet, a new type defined to be CFTypeRef) that can be passed to any function expecting a keychain-or-array. (You can even special-case the len == 1 case, too.) Then write another function that takes a CFTypeRef and releases it and possibly its contents. And then you can just defer that function to do the cleanup.